plugins/node/opentelemetry-instrumentation-aws-sdk/src/aws-sdk.ts

```diff
@@ -0,0 +1,620 @@
+  ) {
+    return function send(
+      this: any,
+      command: V3PluginCommand,
+      ...args: unknown[]
+    ): Promise<any> {
+      command[V3_CLIENT_CONFIG_KEY] = this.config;
+      return original.apply(this, [command, ...args]);
+    };
+  }
+
+  private patchV3MiddlewareStack(
+    moduleVersion: string | undefined,
+    middlewareStackToPatch: MiddlewareStack<any, any>
+  ) {
+    if (!isWrapped(middlewareStackToPatch.resolve)) {
+      this._wrap(
+        middlewareStackToPatch,
+        'resolve',
+        this._getV3MiddlewareStackResolvePatch.bind(this, moduleVersion)
+      );
+    }
+
+    // 'clone' and 'concat' functions are internally calling 'constructStack' which is in same
+    // module, thus not patched, and we need to take care of it specifically.
+    this._wrap(
+      middlewareStackToPatch,
+      'clone',
+      this._getV3MiddlewareStackClonePatch.bind(this, moduleVersion)
+    );
+    this._wrap(
+      middlewareStackToPatch,
+      'concat',
+      this._getV3MiddlewareStackClonePatch.bind(this, moduleVersion)
+    );
+  }
+
+  private _getV3MiddlewareStackClonePatch(
+    moduleVersion: string | undefined,
+    original: (...args: any[]) => MiddlewareStack<any, any>
+  ) {
+    const self = this;
+    return function (this: any, ...args: any[]) {
+      const newStack = original.apply(this, args);
+      self.patchV3MiddlewareStack(moduleVersion, newStack);
+      return newStack;
+    };
+  }
+
+  private _getV3MiddlewareStackResolvePatch(
+    moduleVersion: string | undefined,
+    original: (
+      _handler: any,
+      context: HandlerExecutionContext
+    ) => AwsV3MiddlewareHandler<any, any>
+  ) {
+    const self = this;
+    return function (
+      this: any,
+      _handler: any,
+      awsExecutionContext: HandlerExecutionContext
+    ): AwsV3MiddlewareHandler<any, any> {
+      const origHandler = original.call(this, _handler, awsExecutionContext);
+      const patchedHandler = function (
+        this: any,
+        command: V3PluginCommand
+      ): Promise<any> {
+        const clientConfig = command[V3_CLIENT_CONFIG_KEY];
+        const regionPromise = clientConfig?.region?.();
+        const serviceName =
+          clientConfig?.serviceId ??
+          removeSuffixFromStringIfExists(
+            awsExecutionContext.clientName,
+            'Client'
+          );
+        const commandName =
+          awsExecutionContext.commandName ?? command.constructor?.name;
+        const normalizedRequest = normalizeV3Request(
+          serviceName,
+          commandName,
+          command.input,
+          undefined
+        );
+        const requestMetadata =
+          self.servicesExtensions.requestPreSpanHook(normalizedRequest);
+        const span = self._startAwsV3Span(normalizedRequest, requestMetadata);
+        const activeContextWithSpan = trace.setSpan(context.active(), span);
+
+        const handlerPromise = new Promise(async (resolve, reject) => {
```

Comment:
Looks like another error is that this executor shouldn't be async... can you `await` the regionPromise outside of this executor to avoid making it async?

Reply:
The `_getV3MiddlewareStackResolvePatch` is sync, so I cannot do it there.
What I can do is to convert it from await to Promise.finally, but that would create an ugly code, just to resolve the lint error, which is a bummer

Reply:
ended up refactoring this code to use promise callbacks instead of async/await to calm down the linter.
The code is less idiomatic and harder to read. we might want to refactor this one day